vmdb/app/models/vm_redhat.rb

```diff
@@ -37,4 +37,12 @@ def self.calculate_power_state(raw_power_state)
     else             super
     end
   end
+
+  def supports_migrate?
+    false
+  end
+
+  def supports_reconfigure?
+    false
+  end
```

Comment:
That's good @h-kataria but I can't easily tell what this means `!typ.eql?("clone") && VmOrTemplate.includes_template?(vm_ids)`

It's probably best to move this logic into the model and ask `vm_or_template.supports_reconfigure?`.  As it is now, we don't know which vm_id doesn't support it and we're checking `!typ.eql?("clone")` so we don't even know which operation we're asking for, only what we're NOT asking for.  That seems ripe for problems when we add new things.

Instead of this:
```ruby
      vm_ids = find_checked_items.map(&:to_i).uniq
      if !typ.eql?("clone") && VmOrTemplate.includes_template?(vm_ids)
        render_flash_not_applicable_to_model(typ)
        return
      end
      if typ.eql?("clone") && vm_ids.present? && !VmOrTemplate.cloneable?(vm_ids)
        render_flash_not_applicable_to_model(typ)
        return
      end
      if typ.eql?("publish") && VmOrTemplate.where(:id => vm_ids, :type => %w(VmMicrosoft VmRedhat)).exists?
        render_flash_not_applicable_to_model(typ)
        return
      end
```

I'd prefer the UI code not be knowledge about the backend code.  I'd prefer we do something like this:

```ruby
def prov_redirect(typ=nil)
  ...
  vm_ids = find_checked_items.map(&:to_i).uniq
  return if VmOrTemplate.batch_operation_supported?(vm_ids, typ)
  ...
  render_flash_not_applicable_to_model(typ)
end


# VmOrTemplate
def batch_operation_supported?(ids, operation)
  VmOrTemplate.where(:id => ids).all? { |v| v.public_send("#{operation}_supported?")}
end

# Vm - is clone supported on amazon, redhat, openstack, vmware, microsoft?
def clone_supported?
  false
end

# Vm Vmware/Amazon/Openstack  (is this supported on each?)
def publish_supported?
  true
end


# Template model
def clone_supported?
  false
end

# VM Microsoft/Red Hat
def publish_supported?
  false
end
```

It still feels a bit weird because we're not saying which instances can't do the operation so the user has to figure it out themselves.  

Thoughts @Fryguy ?